scipy/interpolate/_bsplines.py

```diff
@@ -606,6 +607,174 @@ def _process_deriv_spec(deriv):
         ords, vals = [], []
     return np.atleast_1d(ords, vals)
 
+def _woodbury_algorithm(A, ur, ll, b, k):
+    '''
+    Implementation of Woodbury algorithm applied to banded
+    matrices with two blocks in upper right and lower left
+    corners.
+    
+    Parameters
+    ----------
+    A : 2-D array, shape(k, n)
+        Matrix of diagonals of original matrix(see 
+        ``solve_banded`` documentation).
+    ur : 2-D array, shape(bs,bs)
+        Upper right block matrix.
+    ll : 2-D array, shape(bs,bs)
+        Lower left block matrix.
+    b : 1-D array, shape(n,)
+        Vector of constant terms of the SLE.
+    k : int
+        B-spline degree.
+        
+    Returns
+    -------
+    c : 1-D array, shape(n,)
+        Solution of the original SLE.
+        
+    Notes
+    -----
+    SLE - system of linear equations.
+    
+    'n' should be greater than 'k', otherwise corner block
+    elements will intersect with diagonals.
+
+    Almost diagonal matrix with upper right and lower left
+    blocks is solved with the following steps: blocks are
+    represented as vectors and new systems of linear equations
+    are constructed. Left parts of systems are the same - the
+    diagonal elements of the original matrix which makes it
+    possible to use a solution for banded matrices 
+    (``solve_banded``). Then the solution for the original SLE
+    is finally computed.
+
+    References
+    ----------
+    .. [1] William H. Press, Saul A. Teukolsky, William T. Vetterling
+           and Brian P. Flannery, Numerical Recipes, 2007, 2.7.3
+
+    '''
+    bs = int((k-1)/2)
+    n = A.shape[1] + 1
+
+    U = np.zeros((n - 1, k - 1))
+    V = np.zeros((k - 1, n - 1))  # V transpose 
+
+    # upper right
+
+    U[:bs, :bs] = ur
+    for j in range(bs): 
+        V[j, -bs + j] = 1
```

Comment:
minor: can be done with fancy indexing, `V[np.arange(bs), np.arange(bs) - bs]` or some such.